salt/cloud/clouds/nova.py

```diff
@@ -726,12 +726,18 @@ def request_instance(vm_=None, call=None):
 
         else:
             pool = floating_ip_conf.get('pool', 'public')
-            for fl_ip, opts in six.iteritems(conn.floating_ip_list()):
-                if opts['fixed_ip'] is None and opts['pool'] == pool:
-                    floating_ip = fl_ip
-                    break
-            if floating_ip is None:
+            try:
                 floating_ip = conn.floating_ip_create(pool)['ip']
+            except Exception:
+                log.info('A new ip address was unable to be allocated. '
+                         'An ip address will be pulled from the already allocated list, '
+                         'This will cause a race condition when building in parallel.')
```

Comment:
It's sort of unclear what action is being taken here. Is it just informing the user that it *could* fail? If so, shouldn't we just stop execution at this point?